### PR TITLE
Fix: always use origin/master instead of master for homebrew

### LINKED ISSUE #3415: update-homebrew.sh needs to run "git pull" before running " brew bump-formula-pr"
Otherwise the build fails (#3414) and it needs to be ran a second time to actually work.

### PR DESCRIPTION
**Pull request by BYK**
**Summary**

Fixes #3415. The old homebrew update script updated the repo
from remote but did not update the local `master` branch, thus
had the potential for conflicts and mismatches. This patch
forces it to use `origin/master` instead and uses the latest
up-to-date version from the source.

**Test plan**

N/A

### CODE REVIEW

File: scripts/update-homebrew.sh [hunk 1]
```diff
@@ -23,11 +23,10 @@ PATH=$PATH:$HOME/.linuxbrew/bin/
 pushd ~/.linuxbrew/Library/Taps/homebrew/homebrew-core
 #git remote set-url origin https://github.com/Daniel15/homebrew-core # for testing
 git remote set-url origin https://github.com/homebrew/homebrew-core
-git reset --hard HEAD
-git clean -fd
 git fetch --prune origin
+git reset --hard origin/master
+git clean -fd
 # Remove any existing branch (eg. if the previous attempt failed)
-git checkout master
```
Comment (Daniel15) [inline scripts/update-homebrew.sh]: Will `git reset --hard origin/master` switch to the `master` branch? I guess that's why this is not needed?

Could you try switching to a different branch, then running this script, and ensuring it switches to master?

Comment (BYK) [reply scripts/update-homebrew.sh]: Ah, my bad. Didn't realize we had to be on master.

**Comment by BYK** [comment]
@Daniel15 ping :)

**Comment by Daniel15** [comment]
Thanks! Let's see if this makes the automated update work next time :)

**Pull request merged**